container_api/src/container.rs

```diff
@@ -191,49 +189,60 @@ impl Container {
         Ok(ipc_binding)
     }
 
-    fn instance_network_config(
-        &self,
-        net_config: &NetworkConfig,
-    ) -> Result<JsonString, HolochainError> {
-        match self.network_ipc_uri {
-            Some(ref uri) => Ok(JsonString::from(json!(
-                {
-                    "backend_kind": "IPC",
-                    "backend_config": {
-                        "socketType": "zmq",
-                        "bootstrapNodes": net_config.bootstrap_nodes,
-                        "ipcUri": uri.clone()
-                    }
-                }
-            ))),
-            None => Ok(JsonString::from(P2pConfig::unique_mock_config())),
-        }
+    fn instance_p2p_config(&self) -> Result<JsonString, HolochainError> {
+        let config = self.p2p_config.clone().unwrap_or_else(|| {
+            // This should never happen, but we'll throw out a named mock network rather than crashing,
+            // just to be nice (TODO make proper logging statement)
+            println!("warn: instance_network_config called before p2p_config initialized! Using default mock network name.");
+            JsonString::from(P2pConfig::named_mock_config("container-default-mock"))
+        });
+        Ok(config)
     }
 
-    /// Tries to create all instances configured in the given Configuration object.
-    /// Calls `Configuration::check_consistency()` first and clears `self.instances`.
-    /// @TODO: clean up the container creation process to prevent loading config before proper setup,
-    ///        especially regarding the signal handler.
-    ///        (see https://github.com/holochain/holochain-rust/issues/739)
-    pub fn load_config(&mut self) -> Result<(), String> {
-        let _ = self.config.check_consistency()?;
-
-        // if there's no NetworkConfig we won't spawn a network process
-        // and instead configure instances to use the mock network
-        if self.config.network.is_some() {
+    fn initialize_p2p_config(&mut self) -> JsonString {
+        match self.config.network.clone() {
             // if there is a config then either we need to spawn a process and get the
             // ipc_uri for it and save it for future calls to `load_config`
             // or we use that uri value that was created from previous calls!
-            if self.network_ipc_uri.is_none() {
-                self.network_ipc_uri = self
+            Some(ref net_config) => {
+                let uri = self
                     .config
                     .clone()
                     .network
                     .unwrap() // unwrap safe because of check above
                     .n3h_ipc_uri
                     .clone()
                     .or_else(|| self.spawn_network().ok());
+                JsonString::from(json!(
```

Comment:
can we make this a struct in core types?

Reply:
I just moved the json! from one place to another, let's deal with it in another PR. I made an issue: https://github.com/holochain/holochain-rust/issues/841